core/arch/riscv64/riscv64.asm

```diff
@@ -59,7 +59,25 @@ GLOBAL_LABEL(cpuid_supported:)
  */
         DECLARE_FUNC(call_switch_stack)
 GLOBAL_LABEL(call_switch_stack:)
-/* FIXME i#3544: Not implemented */
+        addi    sp, sp, -24 /* Init the stack */
+        sd      ra, 16 (sp)
+        sd      s0, 8 (sp)
+        sd      s1, 0 (sp)
+        /* Check mutex_to_free. */
+        beqz     ARG4, call_dispatch_alt_stack_no_free
+        /* Release the mutex. */
+        addi     ARG4,x0,0
+call_dispatch_alt_stack_no_free:
+        mv       s0, ARG5
+        addi     t0, sp, 0    /* Save the current stack to temporary register */
+        addi     sp, ARG2, 0 /* Move the stack */
+        jr       ARG3
+        addi     sp, t0, 0
+        beqz     s0, GLOBAL_LABEL(unexpected_return)
+        ld       s1, 0 (sp)
+        ld       s2, 8 (sp)
```

Comment:
Shouldn't we be loading s0 here? Since that's what's we stored at `8 (sp)` above

Reply:
This is a small mistake. Fixed.